hudi-common/src/main/java/org/apache/hudi/avro/AvroSchemaCache.java

```diff
@@ -33,7 +35,12 @@ public class AvroSchemaCache {
 
 
   // Ensure that there is only one variable instance of the same schema within an entire JVM lifetime
-  private static final LoadingCache<Schema, Schema> SCHEMA_CACHE = Caffeine.newBuilder().weakValues().maximumSize(1024).build(k -> k);
+  private static final LoadingCache<Schema, Schema> SCHEMA_CACHE = Caffeine.newBuilder().weakValues().maximumSize(1024).build(inputSchema -> {
+    if (inputSchema.equals(HoodieMetadataRecord.getClassSchema())) {
```

Comment:
Not sure why we need this special handling?

Reply:
The `#intern` is low frequency action, there is no much agains here, let's keep the cache working in high-level sense.

Reply:
I don't think we need that, just intern the schema on the fly where necessary because it's low frequency invocations.